### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by dlogushiv**
Tasks 7-1 and 7-2 were completed

### CODE REVIEW

File: src/module7/homework/Main.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        orders.add(order10);
+
+//        System.out.println(users);
+//        System.out.println(orders);
+
+        // sort list by Order price in decrease order
+        Comparator<Order> comparatorOrderPriceDecr = new Comparator<Order>() {
+            @Override
+            public int compare(Order o1, Order o2) {
+                return o2.getPrice() - o1.getPrice();
```
Comment (Jqute) [inline src/module7/homework/Main.java]: compare method should always return:
- 1 if o1 more than o2
- 0 if o1 equals o2
- -1 if o1 less than o2

File: src/module7/homework/Main.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            }
+        };
+        orders.sort(comparatorOrderPriceDecr);
+        System.out.println("Decrease sorting by Order price:\n" + orders);
+
+        // sort list by Order price in increase order AND User city
+        Comparator<Order> comparatorOrderPriceIncr = new Comparator<Order>() {
+            @Override
+            public int compare(Order o1, Order o2) {
```
Comment (Jqute) [inline src/module7/homework/Main.java:75]: the same as above

File: src/module7/homework/Main.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                return res;
+            }
+        };
+        orders.sort(comparatorOrderPriceIncr);
+        System.out.println("\nIncrease sorting by Order price AND User city:\n" + orders);
+
+        // sort list by Order itemName AND ShopIdentificator AND User city
+        Comparator<Order> comparatorThreePoints = new Comparator<Order>() {
+            @Override
+            public int compare(Order o1, Order o2) {
```
Comment (Jqute) [inline src/module7/homework/Main.java:87]: the same as above. Use Integer.compare() to compare **int**s

File: src/module7/homework/Main.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            public int compare(User o1, User o2) {
+                return o1.getCity().compareTo(o2.getCity());
+            }
+        };
+        Set<User> set1 = new TreeSet<>(comparatorByCity);   // initialization of sorted set which contain unique cities using comparatorByCity
+        set1.addAll(users);                                 // get set with unique cities
+        List<User> uniqueCities = new ArrayList<>(set1);    // converting set to list for possibility to get unique city
+        int i = 0;
+        while (i <= uniqueCities.size() - 1) {              // for every unique city
+            List<User> temp = new ArrayList<>(0);           // create new temporary list
```
Comment (Jqute) [inline src/module7/homework/Main.java]: temp list is local variable and we don't get the list of lists as result

File: src/module7/homework/MainTask3.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        orderSet.add(order7);
+        orderSet.add(order8);
+        orderSet.add(order9);
+        orderSet.add(order10);
+//        System.out.println(orderSet);
+
+        // check if set contain Order where User’s lastName is - “Petrov”
+        boolean flag1 = false;
+        for (Order order : orderSet) {
+            if (order.getUser().getLastName() == "Petrov") {
```
Comment (Jqute) [inline src/module7/homework/MainTask3.java]: You should never compare strings with '=='. String is an object, so you compare hashcodes.

File: src/module7/homework/MainTask3.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                flag1 = true;
+                break;
+            }
+        }
+        if (flag1) System.out.println("Set contain Order where User’s lastName is - “Petrov”");
+        else System.out.println("Set don't contain Order where User’s lastName is - “Petrov”");
+
+        // print Order with largest price using only one set method - get
+        System.out.println("\nThe Order with largest price:\n" + orderSet.last());
```
Comment (Jqute) [inline src/module7/homework/MainTask3.java:56]: You print Order with largest price. Are you sure that the last order in the largest?

Comment (dlogushiv) [reply src/module7/homework/MainTask3.java:56]: I'm sure. An orderSet is TreeSet which sorted by order price.
public class Order implements Comparable<Order>
@Override
    public int compareTo(Order o) {
        return this.getPrice()-o.getPrice();
    }

**Comment by Jqute** [comment]
LGTM

**Pull request merged**